Kickstarter-iOS/AppDelegateViewModel.swift

```diff
@@ -276,12 +276,16 @@ public final class AppDelegateViewModel: AppDelegateViewModelType, AppDelegateVi
 
     // Push notifications
 
-    let pushNotificationsPreviouslyAuthorized = self.applicationLaunchOptionsProperty.signal
-      .flatMap { _ in AppEnvironment.current.pushRegistrationType.hasAuthorizedNotifications() }
+    let pushNotificationsPreviouslyAuthorizedAndRegistered = self.applicationLaunchOptionsProperty.signal
+      .flatMap { _ in
+        AppEnvironment.current.pushRegistrationType.registerForPushTokenOnAppLaunch()
```

Comment:
Does this get called on app launch even for people that haven't explicitly opted into notifications? That's what it looks like to me, though I could definitely be wrong since I'm not very familiar with this code. If it's getting called before people opt in to notifications, that's a problem.

Reply:
it's a good call out @ifosli . 

So, after clearing and resetting my simulator and testing on my physical device, I'm now seeing that `UIApplication.shared.registerForRemoteNotifications()` is already being called consistently in our existing `PushRegistration.register(for)` method on app launch when logged in or logged out. That is if the user has previously opted in (expected behavior).

Seeing this new behavior shows that this solution isn't going to solve the initial problem.

Next steps should be to update the current ticket, but we'll need to keep digging to determine why Tarveen and E aren't getting PNs.